File: Leetcode-Problems/Baseball Game.cpp
class Solution
{
public:
    int calPoints(vector<string> &operations)
    {
        vector<int> temp;
        for (string s : operations)
        {
            if (s == "C")
                temp.pop_back();
            else if (s == "D")
                temp.push_back(temp.back() * 2);
            else if (s == "+")
                temp.push_back(temp[temp.size() - 1] + temp[temp.size() - 2]);
            else
                temp.push_back(stoi(s));
        }
        int ans = accumulate(temp.begin(), temp.end(), 0);
        return ans;
    }
};

File: Leetcode-Problems/Baseball Game.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> st = new Stack<>();
        for (String s : operations) {
            if (s.equals("C")) {
                st.pop();
            } else if (s.equals("D")) {
                int t = st.peek();
                st.push(t * 2);
            } else if (s.equals("+")) {
                int a = st.peek();
                st.pop();
                int b = st.peek();
                st.push(a);
                st.push(a + b);
            } else {
                int a = Integer.parseInt(s);
                st.push(a);
            }
        }
        int ans = 0;
        while (!st.isEmpty())
            ans += st.pop();
        return ans;
    }
}